src/linter.cc

```diff
@@ -35,7 +36,7 @@
 namespace zetasql::linter {
 
 // This will eventually be erased.
-absl::Status PrintASTTree(absl::string_view sql) {
+absl::Status PrintASTTree(absl::string_view sql, LinterResult* result) {
```

Comment:
Prefer to return the result instead of creating an output parameter: https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

You can have, for instance, `zetasql_base::StatusOr<LinterResult> PrintASTTree(...)`

Reply:
Yes, makes more sense

Reply:
I think the code was more clean with result passing as a parameter. ZetaSQL parser and formatter is also using this way. As an another option we can only return LinterResult, and make the status part of the LinterResult.

Reply:
ZetaSQL API is not following current best practices :) 
I like returning LinterResult that contains status if the check failed completely. But then each check will have to implement special handling of each function returning absl::Status instead of using RETURN_IF_ERROR macro.

Reply:
Changed all checks to return LinterResult, now tests are also remained simple since .ok() function is also defined in LinterResults.